crates/cheatnet/src/state.rs

```diff
@@ -178,6 +180,7 @@ impl NotEmptyCallStack {
 
 pub struct TraceData {
     pub current_call_stack: NotEmptyCallStack,
+    pub stack_of_resources_used_before_calls: Vec<ExecutionResources>,
```

Comment:
Maybe we could make a new type with `stack` in name for this instead of this long field?

Reply:
I can make a type alias but imo the name is fine

Reply:
Idk how a new type solves this issue though, I would still name it the same way even if it had a separate type. Can just call it `resources_stack`, but imo this verbose name signalises what is in this variable pretty well. Also it is not used in a lot of places, so it won't pollute the code

Reply:
Will do as Maks said